designer/server/src/common/components/pagination/template.test.js

```diff
@@ -0,0 +1,580 @@
+import { renderMacro } from '~/test/helpers/component-helpers.js'
+
+describe('Pagination Component', () => {
+  /** @type {Element | null} */
+  let $pagination = null
+
+  describe('With pagination data', () => {
+    describe('Standard pagination', () => {
+      beforeEach(() => {
+        renderMacro('appPagination', 'pagination/macro.njk', {
+          params: {
+            baseUrl: '/library',
+            pagination: {
+              page: 2,
+              perPage: 10,
+              totalPages: 5,
+              totalItems: 50,
+              pages: [
+                {
+                  number: '1',
+                  href: '/library?page=1&perPage=10',
+                  current: false
+                },
+                {
+                  number: '2',
+                  href: '/library?page=2&perPage=10',
+                  current: true
+                },
+                {
+                  number: '3',
+                  href: '/library?page=3&perPage=10',
+                  current: false
+                },
+                { ellipsis: true },
+                {
+                  number: '5',
+                  href: '/library?page=5&perPage=10',
+                  current: false
+                }
+              ]
+            }
+          }
+        })
+
+        $pagination = document.querySelector('.govuk-pagination')
```

Comment:
Any chance we can use the accessibility-aware [`ByRole` queries](https://testing-library.com/docs/queries/byrole/)?

```suggestion
        $pagination = container.getByRole('navigation', {
          name: 'Pagination'
        })
```